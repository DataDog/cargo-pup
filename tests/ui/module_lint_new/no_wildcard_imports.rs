//@compile-flags: --crate-name test_no_wildcard_imports
//@compile-flags: --crate-type lib

// Specific imports (allowed)
use std::fmt::Display;
use std::collections::HashMap;
use std::io::Read;

// Wildcard imports (should trigger errors)
use std::fmt::*; //~ ERROR: Wildcard imports are not allowed
use std::collections::*; //~ ERROR: Wildcard imports are not allowed

// Nested modules with wildcard imports
pub mod inner_module {
    // Specific import (allowed)
    use std::sync::Arc;
    
    // Wildcard import (now checked by the linter in nested modules)
    use std::sync::*; //~ ERROR: Wildcard imports are not allowed
} 