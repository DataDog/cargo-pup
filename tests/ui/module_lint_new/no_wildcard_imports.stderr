error: Wildcard imports are not allowed
  --> tests/ui/module_lint_new/no_wildcard_imports.rs:10:1
   |
LL | use std::fmt::*;
   | ^^^^^^^^^^^^^^^^
   |
   = help: Import specific items instead of using a wildcard
   = note: Applied by cargo-pup rule 'no_wildcard_imports_test'.
   = note: `#[deny(module_wildcard_import)]` on by default

error: Wildcard imports are not allowed
  --> tests/ui/module_lint_new/no_wildcard_imports.rs:11:1
   |
LL | use std::collections::*;
   | ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: Import specific items instead of using a wildcard
   = note: Applied by cargo-pup rule 'no_wildcard_imports_test'.

error: Wildcard imports are not allowed
  --> tests/ui/module_lint_new/no_wildcard_imports.rs:19:5
   |
LL |     use std::sync::*;
   |     ^^^^^^^^^^^^^^^^^
   |
   = help: Import specific items instead of using a wildcard
   = note: Applied by cargo-pup rule 'no_wildcard_imports_test'.

error: aborting due to 3 previous errors

